src/promptflow-evals/samples/EvaluatorUpload.ipynb

```diff
@@ -0,0 +1,497 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "918d39c2-dfb1-4660-b664-1830adf2708c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def fix_evaluator_code(flex_dir: str) -> None:\n",
+    "    \"\"\"\n",
+    "    Read the flow.flex.yaml file from the flex_dir and remove the line, modifying __path__.\n",
+    "\n",
+    "    :param flex_dir: The directory with evaluator saved in flex format.\n",
+    "    \"\"\"\n",
+    "    with open(os.path.join(flex_dir, 'flow.flex.yaml')) as fp:\n",
+    "        flex_definition = yaml.safe_load(fp)\n",
+    "    entry_script = flex_definition['entry'].split(':')[0] + '.py'\n",
+    "    if entry_script == '__init__.py':\n",
+    "        with open(os.path.join(flex_dir, entry_script)) as f:\n",
+    "            with open(os.path.join(flex_dir, '_' + entry_script), 'w') as new_f:\n",
+    "                for line in f:\n",
+    "                    if not '__path__' in line:\n",
+    "                        new_f.write(line)\n",
+    "        os.replace(os.path.join(flex_dir, '_' + entry_script), os.path.join(flex_dir, entry_script))\n",
+    "\n",
+    "fix_evaluator_code('f1_score')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88fbd6bf-6977-457b-aa55-f6dad9ec1f73",
+   "metadata": {},
+   "source": [
+    "Now let us test the flow with the simple dataset, consisting of one ground true and one actual sentense."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fb73f698-7cab-4b30-8947-411c2060560c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = pd.DataFrame({\n",
+    "    \"ground_truth\": [\"January is the coldest winter month.\"],\n",
+    "    \"answer\": [\"June is the coldest summer month.\"]\n",
+    "})\n",
+    "in_file = 'data.jsonl'\n",
+    "data.to_json('data.jsonl', orient='records', lines=True, index=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd3589c8-0df8-489f-b5a8-beb5ae2aec6a",
+   "metadata": {},
+   "source": [
+    "Load the evaluator in a FLEX format and test it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "841d6109-23b9-45c5-b709-b588f932f29d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "flow_result = pf.test(flow='f1_score', inputs='data.jsonl')\n",
+    "print(f\"Flow outputs: {flow_result}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4a94aab5-73f8-4c7d-a7e0-a92853db0198",
+   "metadata": {},
+   "source": [
+    "Now we have all the tools to upload our model to Azure\n",
+    "### Uploading data to Azure\n",
+    "First we will need to authenticate to azure. For this purpose we will use the the configuration file of the net structure.\n",
+    "```json\r\n",
+    "{\r\n",
+    "    \"resource_group_name\": \"resource-group-name\",\r\n",
```

Comment:
what would happen if use both `registry_name` and `workkspace_name` to initialize the `ml_client` and call `ml_client.evaluators.create_or_update(eval)`.

Reply:
In this case, the validation error will be raised:
```
--> 166     raise ValidationException(
    167         message="Both workspace_name and registry_name cannot be used together, for the ml_client.",
    168         no_personal_data_message="Both workspace_name and registry_name are used for ml_client.",
    169         target=ErrorTarget.GENERAL,
    170         error_category=ErrorCategory.USER_ERROR,
    171     )
```